### PR TITLE
Update github actions to latest versions and add more perl versions

### PR DESCRIPTION
**Pull request by yewtc**
I was assigned this repo as part of the [pull request club](https://pullrequest.club).  I found a few issues. In the end I've decided to do  multiple small pull requests and/or report issues.  This is the first PR and is a simple update to the github action to update some actions to the latest versions to remove deprecation errors and add more perl versions.

### CODE REVIEW

**Comment by hakonhagland** [comment]
> I found a few issues. In the end I've decided to do multiple small pull requests and/or report issues. 

That's great. I have been thinking about fixing things here for a long time but had not had time yet.

**Comment by hakonhagland** [comment]
> In the end I've decided to do multiple small pull requests and/or report issues.

@yewtc I also got this PR https://github.com/hakonhagland/math--gsl/pull/1 some months ago, if you have time to have a look at it, it would be great. (It was submitted to the wrong fork, so it would have to be moved over here)

**Comment by hakonhagland** [comment]
> perl: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by perl)

See: https://github.com/hakonhagland/perl-math-gsl/actions/runs/7050216973/job/19192285221?pr=4#step:8:1494

There are some failing GitHub action checks that we need to address before this can be merged

**Comment by hakonhagland** [comment]
See line 119 in actions.yml:

https://github.com/hakonhagland/perl-math-gsl/blob/31bfffad8ab4e0faafda6a2f3594d7eb74cd463b/.github/workflows/actions.yml#L119

maybe the `GLIBC_2.33 not found error` is related to using ubuntu 20.04, should we try with `ubuntu-latest` instead?

**Comment by yewtc** [comment]
This PR was really just to do some housekeeping  with the github actions and not address the more substantial issues. It seemed to me that adding perl versions and updating the action versions was uncontentious - but I agree it still fails. My other PR  points to fixing the more substantial issue - admittedly without fixing it :-)

**Comment by hakonhagland** [comment]
> This PR was really just to do some housekeeping with the github actions and not address the more substantial issues.

Ok let's merge this PR and fix the failed checks later.

**Comment by hakonhagland** [comment]
> perl: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by perl)

Added #6 to try to fix this issue

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -1,17 +1,17 @@
 name: linux-build-dist
-on: [push, pull_request]
+on: [push, pull_request, workflow_dispatch]
```
Comment (hakonhagland) [inline .github/workflows/actions.yml:2]: I wonder what `workflow_dispatch` do here? It seems if I submit a PR, the workflow is now triggered twice, once for the "pull_request" and once for the "workflow_dispatch"

Comment (yewtc) [reply .github/workflows/actions.yml:2]: It should just mean that a button appears in the actions tabs that mean you can run the action 'manually'. It shouldn't do anything automatically.

Comment (hakonhagland) [reply .github/workflows/actions.yml:2]: > a button appears in the actions tabs that mean you can run the action 'manually'

Ok, thanks!

Comment (hakonhagland) [reply .github/workflows/actions.yml:2]: > It seems if I submit a PR, the workflow is now triggered twice, once for the "pull_request" and once for the "workflow_dispatch"

I'll try to fix this by #7

**Pull request merged**